rest/url_blackbox_test.go

```diff
@@ -67,3 +67,44 @@ func TestReplaceDomainPrefixInTooShortHostFails(t *testing.T) {
 	_, err := ReplaceDomainPrefix("org", "sso")
 	assert.NotNil(t, err)
 }
+
+func TestValidateEmailSuccess(t *testing.T) {
+	resource.Require(t, resource.UnitTest)
+	t.Parallel()
+
+	isValid, err := ValidateEmail("a@a.com")
+	require.NoError(t, err)
+	require.True(t, isValid)
+}
+
+func TestValidateEmailFail(t *testing.T) {
+	resource.Require(t, resource.UnitTest)
+	t.Parallel()
+
+	isValid, err := ValidateEmail("a.a@com")
+	require.NoError(t, err)
+	require.False(t, isValid)
+}
+
+func TestAddParamsSuccess(t *testing.T) {
+	resource.Require(t, resource.UnitTest)
+	t.Parallel()
+
+	testMap := map[string]string{
+		"param1": "a",
+		"param2": "b",
+		"param3": "https://www.redhat.com",
+	}
+	generatedURL, err := AddParams("https://openshift.io", testMap)
+	require.NoError(t, err)
+	assert.Equal(t, "https://openshift.io?param1=a&param2=b&param3=https%3A%2F%2Fwww.redhat.com", generatedURL)
```

Comment:
We can't use equal here. Params are a map. The order is not guaranteed.

Reply:
fixed https://github.com/fabric8-services/fabric8-auth/pull/231/files#diff-7b2650336b2fd942e5bb9a1d8aec902fR99